src/test/java/org/opensearch/performanceanalyzer/rest/QueryRcaRequestHandlerTest.java

```diff
@@ -0,0 +1,144 @@
+ */
+
+package org.opensearch.performanceanalyzer.rest;
+
+
+import com.sun.net.httpserver.Headers;
+import com.sun.net.httpserver.HttpExchange;
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.util.Collections;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+import org.opensearch.performanceanalyzer.AppContext;
+import org.opensearch.performanceanalyzer.commons.metrics.AllMetrics;
+import org.opensearch.performanceanalyzer.rca.framework.core.Stats;
+import org.opensearch.performanceanalyzer.reader.ClusterDetailsEventProcessor;
+
+public class QueryRcaRequestHandlerTest {
+    private QueryRcaRequestHandler handler;
+    private AppContext appContext;
+    private final String queryPrefix = "http://localhost:9600/_plugins/_performanceanalyzer/rca";
+
+    private void setClusterManagerContext(boolean isClusterManager) {
+        ClusterDetailsEventProcessor clusterDetailsEventProcessor =
+                new ClusterDetailsEventProcessor();
+        clusterDetailsEventProcessor.setNodesDetails(
+                Collections.singletonList(
+                        new ClusterDetailsEventProcessor.NodeDetails(
+                                isClusterManager
+                                        ? AllMetrics.NodeRole.ELECTED_CLUSTER_MANAGER
+                                        : AllMetrics.NodeRole.DATA,
+                                "test_node",
+                                "127.0.0.1",
+                                isClusterManager)));
+        appContext.setClusterDetailsEventProcessor(clusterDetailsEventProcessor);
+    }
+
+    private HttpExchange sendQuery(String query, String requestMethod, OutputStream os)
+            throws Exception {
+        HttpExchange exchange = Mockito.mock(HttpExchange.class);
+        Mockito.when(exchange.getResponseBody()).thenReturn(os != null ? os : System.out);
+        Mockito.when(exchange.getRequestMethod()).thenReturn(requestMethod);
+        Headers responseHeaders = Mockito.mock(Headers.class);
+        Mockito.when(exchange.getResponseHeaders()).thenReturn(responseHeaders);
+        Mockito.when(exchange.getRequestURI()).thenReturn(new URI(query));
+        handler.handle(exchange);
+        return exchange;
+    }
+
+    @Before
+    public void setUp() {
+        appContext = new AppContext();
+        handler = new QueryRcaRequestHandler(appContext);
+        setClusterManagerContext(false);
+        Stats.getInstance().getConnectedComponents(); // Initializes muted graph nodes
+    }
+
+    @After
+    public void undo() {
+        Stats.getInstance().getConnectedComponents();
+    }
+
+    @Test
+    public void testInvalidNodeRole() throws Exception {
+        HttpExchange exchange = sendQuery(queryPrefix + "?name=HotShardClusterRca", "GET", null);
```

Comment:
Before sending the request, we should assert `HotShardClusterRca` is part of the Muted RCA in the component graph.